packages/alea-frontend/pages/study-buddy/index.tsx

```diff
@@ -50,7 +61,91 @@ function removeRecentCourse(courseCode: string) {
     );
   }
 }
+function StudyBuddyOverviewGrap(){
+  const [sortedCoureses,setSortedCoureses]=useState<GetSortedCoursesByConnectionStudybody[]>();
+  const [selectedCourseIndex, setSelectedCourseIndex] = useState<string>(null);
+  const [connections, setConnections] = useState<UserStats['connections']>([]);
+  const [userIdsAndActiveStatus, setUserIdsAndActiveStatus] = useState([]);
+  useEffect(()=>{
+    const fetchData= ()=>{
+      getStudyBodyCouresesSortedbyConnections().then(setSortedCoureses);
+    };
+    fetchData();
+  },[]);
+  const handleListItemClick = async (
+    courseId: string,
+  ) => {
+    setSelectedCourseIndex(courseId);
+    const data = await getStudyBuddyUsersStats(courseId);
+    setConnections(data.connections);
+    setUserIdsAndActiveStatus(data.userIdsAndActiveStatus)
+  };
+  const courseList=sortedCoureses?.map((c,i)=>(
+    <ListItemButton key={i}
+    selected={selectedCourseIndex === c.courseId}
+    onClick={() => handleListItemClick(c.courseId)}>
+    <ListItemText primary={(MaAI_COURSES[c.courseId]?.courseName ?? c.courseId)+` (${c.member})`} />
+  </ListItemButton>
+))
+  return ( <>         
+  <List>
+    {courseList}
+    
+  </List>
+ {selectedCourseIndex===null?null:<StudyBuddyConnectionsGraph
+            connections={connections}
+            userIdsAndActiveStatus={userIdsAndActiveStatus}
+          /> } 
+          </>)
+}
+function StatsForModerator() {
```

Comment:
Instead of copy-paste, can we move this to a separate component file and then it can be imported in both places where it is needed